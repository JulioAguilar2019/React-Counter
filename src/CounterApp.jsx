import PropTypes from 'prop-types';
import { useState } from 'react'





export const CounterApp = ({ value }) => {


    console.log('render');

    const [counter, setCounter] = useState(value);

    const handleSuma = () => {
        setCounter(counter + 1);
    }
    

    const handleResta = () => {
        setCounter(counter - 1);
    }

    const handleReset = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>Counter app</h1>
            <h2> {counter} </h2>
            <button onClick={ handleSuma }>Sumar 1 al counter</button>
            <button onClick={ handleResta }>Restar 1 al counter</button>
            <button onClick={ handleReset }>Resetear el counter</button>
            
        </>
    )
};

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
};

CounterApp.defaultProps = {
    value: 10
}

