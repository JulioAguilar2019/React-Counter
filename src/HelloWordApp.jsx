// const newMesagge = {
//     message: 'Julio',
//     tittle: 'Es un titulo'
// };

// const GetName = (name) => {
//     return name;
// }
import PropTypes from 'prop-types'

export const HelloWordApp = ({ title, subTitle, name }) => {

    return (
        <>
            <h1 data-testid="test-title"> {title} </h1>
            {/* <code> {JSON.stringify(newMesagge)}</code> */}
            <p>{subTitle}</p>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>

    )
}


HelloWordApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}


HelloWordApp.defaultProps = {
    // title: "no hay titulo",
    subTitle: 'no hay subtitulo',
    name: 'no hay nombre'
}