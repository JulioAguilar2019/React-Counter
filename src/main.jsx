import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
//import { FirstApp } from './FirstApp';
import { HelloWordApp } from './HelloWordApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
       {/* <HelloWordApp title= 'este es el titulo' /> */}
    <CounterApp/>
    </React.StrictMode>
)

