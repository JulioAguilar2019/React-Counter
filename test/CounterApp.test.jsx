
import { fireEvent, render, screen } from "@testing-library/react"
import { CounterApp } from "../src/CounterApp"

describe('Estas son las pruebas para el counter app', () => {

    const initValue = 10;

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<CounterApp />)

        expect(container).toMatchSnapshot();

    });

    test('Debe de mostrar el valor inicial de 100', () => {

        render(<CounterApp value={100} />)

        expect(screen.getByText(100)).toBeTruthy();
    });

    test('Debe de incrementar con el boton sumar 1', () => {

        render(<CounterApp value={ initValue } />);
        fireEvent.click(screen.getByText('Sumar 1 al counter'))
        expect(screen.getByText('11') ).toBeTruthy();
    });

    test('Debe de restar con el boton restar 1', () => {

        render(<CounterApp value={ initValue } />);
        fireEvent.click(screen.getByText('Restar 1 al counter'))
        expect(screen.getByText('9') ).toBeTruthy();
    });

    test('Debe de resetear el counter con el boton resetear counter', () => {

        render(<CounterApp value={ initValue } />);
        fireEvent.click(screen.getByText('Restar 1 al counter'))
        fireEvent.click(screen.getByText('Restar 1 al counter'))
        // fireEvent.click(screen.getByText('Resetear el counter'))
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        expect(screen.getByText('10') ).toBeTruthy();
        
    });


});