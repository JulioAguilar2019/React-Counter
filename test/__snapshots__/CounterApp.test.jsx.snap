// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Estas son las pruebas para el counter app Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter app
  </h1>
  <h2
    data-testid="counterId"
  >
     
    10
     
  </h2>
  <button>
    Sumar 1 al counter
  </button>
  <button>
    Restar 1 al counter
  </button>
  <button
    aria-label="btn-reset"
  >
    Resetear el counter
  </button>
</div>
`;
